### PR TITLE
Fix workflow controller watcher and rbac

### LINKED ISSUE #146: BUG: "system:serviceaccount:addon-manager-system:default" cannot list resource "workflows" in API group "argoproj.io" at the cluster scope
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG
**What happened**:
```
E0602 23:56:40.972218       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Workflow: failed to list *v1alpha1.Workflow: workflows.argoproj.io is forbidden: User "system:serviceaccount:addon-manager-system:default" cannot list resource "workflows" in API group "argoproj.io" at the cluster scope
E0602 23:57:21.057596       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Workflow: failed to list *v1alpha1.Workflow: workflows.argoproj.io is forbidden: User "system:serviceaccount:addon-manager-system:default" cannot list resource "workflows" in API group "argoproj.io" at the cluster scope
E0602 23:58:13.462508       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/reflector.go:167: Failed to watch *v1alpha1.Workflow: failed to list *v1alpha1.Workflow: workflows.argoproj.io is forbidden: User "system:serviceaccount:addon-manager-system:default" cannot list resource "workflows" in API group "argoproj.io" at the cluster scope
{"level":"error","ts":1654214325.9899297,"logger":"controller-runtime.manager.controller.addon-manager-wf-controller","msg":"Could not wait for Cache to sync","error":"failed to wait for addon-manager-wf-controller caches to sync: timed out waiting for cache to be synced","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:195\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:221\nsigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/manager/internal.go:696"}
{"level":"error","ts":1654214325.990036,"logger":"controller-runtime.manager.controller.addon-manager-controller","msg":"Could not wait for Cache to sync","error":"failed to wait for addon-manager-controller caches to sync: timed out waiting for cache to be synced","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:195\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/internal/controller/controller.go:221\nsigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.2/pkg/manager/internal.go:696"}
{"level":"error","ts":1654214325.9905198,"logger":"controller-runtime.manager","msg":"error received after stop sequence was engaged","error":"failed to wait for addon-manager-controller caches to sync: timed out waiting for cache to be synced"}
{"level":"error","ts":1654214325.9906044,"logger":"setup","msg":"problem running manager","error":"failed to wait for addon-manager-wf-controller caches to sync: timed out waiting for cache to be synced","stacktrace":"main.main\n\t/workspace/main.go:71\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:255"}
```
**What you expected to happen**:
This is a regression from when we limited RBAC to namespace for Argo WF. https://github.com/keikoproj/addon-manager/pull/73/files#diff-2df6a56a33260adb76f704ff7c6737be9f43c13fe523521ee76390ff2a9ca4b8R151

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Addon Manager version
- Kubernetes version :
```
$ kubectl version -o yaml
```

**Other debugging information (if applicable)**:
- Addon status:
```
$ kubectl describe addon <addon-name>
```
- controller logs:
```
$ kubectl logs <addon-manager-pod>
```

### PR DESCRIPTION
**Pull request by kevdowney**
Fixes #146 

Limit the watch for Workflows to be namespaced

Signed-off-by: kevdowney <kevdowney@gmail.com>

### CODE REVIEW

**Review by shailshah9 — APPROVED** [review]

**Review by figo — DISMISSED** [review]

**Review by shaoxt — APPROVED** [review]
LGTM

File: controllers/controller_manager_setup.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"github.com/keikoproj/addon-manager/pkg/common"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func New(mgr manager.Manager) error {
+	versionCache := addon.NewAddonVersionCacheClient()
+	dynClient := dynamic.NewForConfigOrDie(mgr.GetConfig())
+	nsInformers := dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynClient, addonapiv1.AddonResyncPeriod, addonapiv1.ManagedNameSpace, nil)
```
Comment (ccfishk) [inline controllers/controller_manager_setup.go:18]: Only the managed namespace ? 
The namespace of each of the addons would be watched by the controller also.

Comment (kevdowney) [reply controllers/controller_manager_setup.go:18]: This is only for the WorkflowInformer below, so it's restriced to addon-manager-system ns.

Comment (ccfishk) [reply controllers/controller_manager_setup.go:18]: I would assuming the watched namespace solution could be generic that applied to all resources that add-on manager is watching.

Comment (kevdowney) [reply controllers/controller_manager_setup.go:18]: I tried the manager namespace and it failed to sync caches, that can be looked into later for now we restricted this informer and also gave the controller cluster scope watch permissions just so we can unblock.

**Review by figo — APPROVED** [review]

**Pull request merged**